src/backend/common/deferred/clients/fake_client.py

```diff
@@ -1,4 +1,46 @@
 from backend.common.deferred.clients.rq_client import RQTaskClient
+from backend.common.deferred.queues.rq_queue import RQTaskQueue
+from backend.common.deferred.tasks.task import Task
+from backend.common.redis import RedisClient
+
+
+class InlineRQTaskQueue(RQTaskQueue):
+    """
+    A RQ-backed queue, but will run jobs inline
+    instead of making a HTTP request callback
+    """
+
+    def enqueue(self, task: Task, *args, **kwargs) -> None:
+        self._queue.enqueue(task.obj, *task.args, **task.kwargs)
+
+
+class InlineRQTaskClient(RQTaskClient):
+    def queue(self, name) -> InlineRQTaskQueue:
+        return InlineRQTaskQueue(
+            name, default_service=self.default_service, redis_client=self._client
+        )
+
+
+class InlineTaskClient(InlineRQTaskClient):
```

Comment:
Can we just remove all of this `Inline` stuff and make this how the `FakeTaskClient` works?

I'm also not sure I understand why the double classes here (`InlineRQTaskClient` and `InlineTaskClient`)

Reply:
I did that originally, but it broke all the `deferred` tests that constructed http requests. We can potentially change the default with some more work, but again, that would be better for a separate diff IMO

Reply:
Maybe I can take a pass at it! I'm willing to update tests in order to make this better

Reply:
I think it makes sense as the default behavior, but it is probably work to be done separately from this diff